<a href="https://colab.research.google.com/github/Xavier-ML/BIOESTADISTICA-XAVIER-ML/blob/main/Dia_7_CIBERTEC.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
#web scraping
import requests

In [6]:
# Pandas tarea
import pandas as pd

In [7]:
datos = pd.read_csv("/content/datos.csv", sep = "," )

P1. Cargar el dataset datos.csv en Pandas. Mostrar las filas iniciales iniciales en la pantalla. Cuantas filas y columnas hay.

In [8]:
datos.head(9)

Unnamed: 0  age         workclass  fnlwgt  education  education-num  \
0           0   50  Self-emp-not-inc   83311  Bachelors             13   
1           1   38           Private  215646    HS-grad              9   
2           2   53           Private  234721       11th              7   
3           3   28           Private  338409  Bachelors             13   
4           4   37           Private  284582    Masters             14   
5           5   49           Private  160187        9th              5   
6           6   52  Self-emp-not-inc  209642    HS-grad              9   
7           7   31           Private   45781    Masters             14   
8           8   42           Private  159449  Bachelors             13   

          marital-status         occupation   relationship   race     sex  \
0     Married-civ-spouse    Exec-managerial        Husband  White    Male   
1               Divorced  Handlers-cleaners  Not-in-family  White    Male   
2     Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
3     Married-civ-spouse     Prof-specialty           Wife  Black  Female   
4     Married-civ-spouse    Exec-managerial           Wife  White  Female   
5  Married-spouse-absent      Other-service  Not-in-family  Black  Female   
6     Married-civ-spouse    Exec-managerial        Husband  White    Male   
7          Never-married     Prof-specialty  Not-in-family  White  Female   
8     Married-civ-spouse    Exec-managerial        Husband  White    Male   

   capital-gain  capital-loss  hours-per-week native-country salary  
0             0             0              13  United-States  <=50K  
1             0             0              40  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40           Cuba  <=50K  
4             0             0              40  United-States  <=50K  
5             0             0              16        Jamaica  <=50K  
6             0             0              45  United-States   >50K  
7         14084             0              50  United-States   >50K  
8          5178             0              40  United-States   >50K

In [9]:
datos.shape

(32560, 16)

P2. ¿Cuántas perosnas de sexo masculino y femenino están representadas en el dataset?

In [10]:
grupoMaculino = datos[datos.sex == "Male"]
grupoMaculino.shape

(21789, 16)

In [11]:
grupoMujeres = datos[datos.sex == "Female"]
grupoMujeres.shape

(10771, 16)

In [12]:
21789 + 10771

32560

P3. ¿Cuál es la edad promedio del sexo masculino?

In [13]:
grupoMaculino

Unnamed: 0  age         workclass  fnlwgt     education  education-num  \
0               0   50  Self-emp-not-inc   83311     Bachelors             13   
1               1   38           Private  215646       HS-grad              9   
2               2   53           Private  234721          11th              7   
6               6   52  Self-emp-not-inc  209642       HS-grad              9   
8               8   42           Private  159449     Bachelors             13   
...           ...  ...               ...     ...           ...            ...   
32552       32552   32           Private  116138       Masters             14   
32553       32553   53           Private  321865       Masters             14   
32554       32554   22           Private  310152  Some-college             10   
32556       32556   40           Private  154374       HS-grad              9   
32558       32558   22           Private  201490       HS-grad              9   

           marital-status         occupation   relationship  \
0      Married-civ-spouse    Exec-managerial        Husband   
1                Divorced  Handlers-cleaners  Not-in-family   
2      Married-civ-spouse  Handlers-cleaners        Husband   
6      Married-civ-spouse    Exec-managerial        Husband   
8      Married-civ-spouse    Exec-managerial        Husband   
...                   ...                ...            ...   
32552       Never-married       Tech-support  Not-in-family   
32553  Married-civ-spouse    Exec-managerial        Husband   
32554       Never-married    Protective-serv  Not-in-family   
32556  Married-civ-spouse  Machine-op-inspct        Husband   
32558       Never-married       Adm-clerical      Own-child   

                     race   sex  capital-gain  capital-loss  hours-per-week  \
0                   White  Male             0             0              13   
1                   White  Male             0             0              40   
2                   Black  Male             0             0              40   
6                   White  Male             0             0              45   
8                   White  Male          5178             0              40   
...                   ...   ...           ...           ...             ...   
32552  Asian-Pac-Islander  Male             0             0              11   
32553               White  Male             0             0              40   
32554               White  Male             0             0              40   
32556               White  Male             0             0              40   
32558               White  Male             0             0              20   

      native-country salary  
0      United-States  <=50K  
1      United-States  <=50K  
2      United-States  <=50K  
6      United-States   >50K  
8      United-States   >50K  
...              ...    ...  
32552         Taiwan  <=50K  
32553  United-States   >50K  
32554  United-States  <=50K  
32556  United-States   >50K  
32558  United-States  <=50K  

[21789 rows x 16 columns]

In [14]:
grupoMaculino["age"].mean()

39.433567396392675

P4. ¿Cuál es la promoción de cubanos del dataset?

In [15]:
datos.columns

Index(['Unnamed: 0', 'age', 'workclass', 'fnlwgt', 'education',
       'education-num', 'marital-status', 'occupation', 'relationship', 'race',
       'sex', 'capital-gain', 'capital-loss', 'hours-per-week',
       'native-country', 'salary'],
      dtype='object')

In [16]:
datos["native-country"] == "Cuba"

0        False
1        False
2        False
3         True
4        False
         ...  
32555    False
32556    False
32557    False
32558    False
32559    False
Name: native-country, Length: 32560, dtype: bool

In [18]:
Cubanos = datos[datos["native-country"] == "Cuba"]
Cubanos

Unnamed: 0  age         workclass  fnlwgt  education  education-num  \
3               3   28           Private  338409  Bachelors             13   
80             80   52           Private  276515  Bachelors             13   
637           637   47      Self-emp-inc  248145    5th-6th              3   
700           700   31           Private  243605  Bachelors             13   
1236         1236   41  Self-emp-not-inc  209344    HS-grad              9   
...           ...  ...               ...     ...        ...            ...   
30840       30840   37           Private  238433    1st-4th              2   
30891       30891   52           Private  222646       12th              8   
30905       30905   39  Self-emp-not-inc  251323        9th              5   
31148       31148   52      Self-emp-inc  230767  Bachelors             13   
31906       31906   48           Private  379883    Masters             14   

           marital-status         occupation    relationship   race     sex  \
3      Married-civ-spouse     Prof-specialty            Wife  Black  Female   
80     Married-civ-spouse      Other-service         Husband  White    Male   
637    Married-civ-spouse   Transport-moving         Husband  White    Male   
700               Widowed              Sales       Unmarried  White  Female   
1236   Married-civ-spouse              Sales  Other-relative  White  Female   
...                   ...                ...             ...    ...     ...   
30840  Married-civ-spouse   Transport-moving         Husband  White    Male   
30891           Separated  Machine-op-inspct  Other-relative  White  Female   
30905  Married-civ-spouse    Farming-fishing  Other-relative  White    Male   
31148  Married-civ-spouse    Exec-managerial         Husband  White    Male   
31906  Married-civ-spouse     Prof-specialty         Husband  White    Male   

       capital-gain  capital-loss  hours-per-week native-country salary  
3                 0             0              40           Cuba  <=50K  
80                0             0              40           Cuba  <=50K  
637               0             0              50           Cuba  <=50K  
700               0          1380              40           Cuba  <=50K  
1236              0             0              40           Cuba  <=50K  
...             ...           ...             ...            ...    ...  
30840             0             0              40           Cuba  <=50K  
30891             0             0              40           Cuba  <=50K  
30905             0             0              40           Cuba  <=50K  
31148             0          1902              60           Cuba   >50K  
31906             0             0              40           Cuba   >50K  

[95 rows x 16 columns]

In [19]:
Cubanos.shape
# 95 son Cubanos

(95, 16)

P5. ¿cuál es la edad promedio de aquellos que ganan más de 50K y menos de 50K?

In [20]:
gananmayora50k = datos[datos.salary == ">50K"]
gananmenora50k = datos[datos.salary == "<50K"]

In [21]:
gananmayora50k

Unnamed: 0  age         workclass  fnlwgt     education  education-num  \
6               6   52  Self-emp-not-inc  209642       HS-grad              9   
7               7   31           Private   45781       Masters             14   
8               8   42           Private  159449     Bachelors             13   
9               9   37           Private  280464  Some-college             10   
10             10   30         State-gov  141297     Bachelors             13   
...           ...  ...               ...     ...           ...            ...   
32538       32538   71                 ?  287372     Doctorate             16   
32544       32544   39         Local-gov  111499    Assoc-acdm             12   
32553       32553   53           Private  321865       Masters             14   
32556       32556   40           Private  154374       HS-grad              9   
32559       32559   52      Self-emp-inc  287927       HS-grad              9   

           marital-status         occupation   relationship  \
6      Married-civ-spouse    Exec-managerial        Husband   
7           Never-married     Prof-specialty  Not-in-family   
8      Married-civ-spouse    Exec-managerial        Husband   
9      Married-civ-spouse    Exec-managerial        Husband   
10     Married-civ-spouse     Prof-specialty        Husband   
...                   ...                ...            ...   
32538  Married-civ-spouse                  ?        Husband   
32544  Married-civ-spouse       Adm-clerical           Wife   
32553  Married-civ-spouse    Exec-managerial        Husband   
32556  Married-civ-spouse  Machine-op-inspct        Husband   
32559  Married-civ-spouse    Exec-managerial           Wife   

                     race     sex  capital-gain  capital-loss  hours-per-week  \
6                   White    Male             0             0              45   
7                   White  Female         14084             0              50   
8                   White    Male          5178             0              40   
9                   Black    Male             0             0              80   
10     Asian-Pac-Islander    Male             0             0              40   
...                   ...     ...           ...           ...             ...   
32538               White    Male             0             0              10   
32544               White  Female             0             0              20   
32553               White    Male             0             0              40   
32556               White    Male             0             0              40   
32559               White  Female         15024             0              40   

      native-country salary  
6      United-States   >50K  
7      United-States   >50K  
8      United-States   >50K  
9      United-States   >50K  
10             India   >50K  
...              ...    ...  
32538  United-States   >50K  
32544  United-States   >50K  
32553  United-States   >50K  
32556  United-States   >50K  
32559  United-States   >50K  

[7841 rows x 16 columns]

In [24]:
gananmenora50k

Empty DataFrame
Columns: [Unnamed: 0, age, workclass, fnlwgt, education, education-num, marital-status, occupation, relationship, race, sex, capital-gain, capital-loss, hours-per-week, native-country, salary]
Index: []

P6. Todos los que ganan más de 50k al año tienen master?

In [25]:
gananmayora50k

Unnamed: 0  age         workclass  fnlwgt     education  education-num  \
6               6   52  Self-emp-not-inc  209642       HS-grad              9   
7               7   31           Private   45781       Masters             14   
8               8   42           Private  159449     Bachelors             13   
9               9   37           Private  280464  Some-college             10   
10             10   30         State-gov  141297     Bachelors             13   
...           ...  ...               ...     ...           ...            ...   
32538       32538   71                 ?  287372     Doctorate             16   
32544       32544   39         Local-gov  111499    Assoc-acdm             12   
32553       32553   53           Private  321865       Masters             14   
32556       32556   40           Private  154374       HS-grad              9   
32559       32559   52      Self-emp-inc  287927       HS-grad              9   

           marital-status         occupation   relationship  \
6      Married-civ-spouse    Exec-managerial        Husband   
7           Never-married     Prof-specialty  Not-in-family   
8      Married-civ-spouse    Exec-managerial        Husband   
9      Married-civ-spouse    Exec-managerial        Husband   
10     Married-civ-spouse     Prof-specialty        Husband   
...                   ...                ...            ...   
32538  Married-civ-spouse                  ?        Husband   
32544  Married-civ-spouse       Adm-clerical           Wife   
32553  Married-civ-spouse    Exec-managerial        Husband   
32556  Married-civ-spouse  Machine-op-inspct        Husband   
32559  Married-civ-spouse    Exec-managerial           Wife   

                     race     sex  capital-gain  capital-loss  hours-per-week  \
6                   White    Male             0             0              45   
7                   White  Female         14084             0              50   
8                   White    Male          5178             0              40   
9                   Black    Male             0             0              80   
10     Asian-Pac-Islander    Male             0             0              40   
...                   ...     ...           ...           ...             ...   
32538               White    Male             0             0              10   
32544               White  Female             0             0              20   
32553               White    Male             0             0              40   
32556               White    Male             0             0              40   
32559               White  Female         15024             0              40   

      native-country salary  
6      United-States   >50K  
7      United-States   >50K  
8      United-States   >50K  
9      United-States   >50K  
10             India   >50K  
...              ...    ...  
32538  United-States   >50K  
32544  United-States   >50K  
32553  United-States   >50K  
32556  United-States   >50K  
32559  United-States   >50K  

[7841 rows x 16 columns]

In [34]:
type(gananmayora50k)

pandas.core.frame.DataFrame

In [40]:
Nomaster = gananmayora50k[gananmayora50k["education"] != "Masters"]

In [43]:
Nomaster["education"].count()

6882

In [44]:
Simaster = gananmayora50k[gananmayora50k["education"] == "Masters"]

In [46]:
Simaster["education"].count()

959

In [47]:
6882 + 959

7841

DE LOS 7841 ENTREVISTADOS SOLO 959 SON MASTERS Y LOS 6882 SON NO MASTERS.